src/pages/fieldManagement/settings/defaultFieldComponents/RelationshipEditor.jsx

```diff
@@ -0,0 +1,304 @@
+  return recomposeAllRelationshipObjectFromModifiedCategoryObject(
+    allRelationshipsObj,
+    targetRelationshipCategoryObj,
+  );
+}
+
+function addRelationshipToCategory(
+  allRelationshipsObj,
+  currentCategory,
+  newRelationship,
+) {
+  const targetRelationshipCategoryObj = get(
+    filter(
+      allRelationshipsObj,
+      currentRelationshipCategory =>
+        currentRelationshipCategory.category === currentCategory,
+    ),
+    '0',
+    {},
+  );
+  const targetRelationshipArr =
+    targetRelationshipCategoryObj?.relationships;
+  targetRelationshipArr.push(newRelationship);
+  const modifiedArr = targetRelationshipArr;
+  const modifiedTargetCategoryObj = {
+    category: currentCategory,
+    relationships: modifiedArr,
+  };
+  return recomposeAllRelationshipObjectFromModifiedCategoryObject(
+    allRelationshipsObj,
+    modifiedTargetCategoryObj,
+  );
+}
+
+function renderSingleRelationship(
+  allRelationshipsObj,
+  currentCategory,
+  relationship,
+  onChange,
+) {
+  return (
+    <div style={{ marginLeft: 32, marginTop: 10 }}>
+      <TextInput
+        width={240}
+        schema={relationship}
+        onChange={newRelationshipName => {
+          onChange(
+            updateRelationshipInCategory(
+              allRelationshipsObj,
+              currentCategory,
+              relationship,
+              newRelationshipName,
+            ),
+          );
+        }}
+        value={relationship}
+        autoFocus
+        InputProps={{
+          endAdornment: (
+            <InputAdornment position="end">
+              <DeleteButton
+                onClick={() => {
+                  onChange(
+                    deleteRelationship(
+                      allRelationshipsObj,
+                      currentCategory,
+                      relationship,
+                    ),
+                  );
+                }}
+              />
+            </InputAdornment>
+          ),
+        }}
+      />
+    </div>
+  );
+}
+
+function renderSingleRelationshipObj(
+  allRelationshipsObj,
+  singleRelationshipObj,
+  onChange,
+) {
+  const category = singleRelationshipObj?.category;
+  const relationships = singleRelationshipObj?.relationships;
+  return (
+    <div style={{ marginTop: 10 }}>
+      <TextInput
+        width={240}
+        schema={category}
+        onChange={newName => {
+          onChange(
+            updateCategoryName(
+              allRelationshipsObj,
+              category,
+              newName,
+            ),
+          );
+        }}
+        value={category}
+        autoFocus
+        InputProps={{
+          endAdornment: (
+            <InputAdornment position="end">
+              <IconButton
+                size="small"
+                onClick={() => {
+                  onChange(
+                    addRelationshipToCategory(
+                      allRelationshipsObj,
+                      category,
+                      '',
```

Comment:
We could easily make this something like, "Enter new relationship name here"

Reply:
I would rather leave it blank and add a placeholder if you feel that it needs it.

Reply:
Done